packages/stockflux-launcher/src/App.js

```diff
@@ -1,69 +1,112 @@
-import React, { useRef, useEffect } from 'react';
 import cx from 'classnames';
+import { ScreenEdge, useDockWindow, useOptions } from 'openfin-react-hooks';
+import React, { useEffect, useState } from 'react';
+import Components from 'stockflux-components';
 import { OpenfinApiHelpers } from 'stockflux-core';
-import { useDockWindow, ScreenEdge } from 'openfin-react-hooks';
-import {
-  FaChevronUp,
-  FaChevronLeft,
-  FaChevronRight,
-  FaRegHandRock
-} from 'react-icons/fa';
-import AppShortcuts from './app-shortcuts/AppShortcuts';
+import './App.css';
 import FreeTextSearch from './free-text-search/FreeTextSearch';
+import getUndockedPosition from './helpers/getUndockedPosition';
+import { ReactComponent as LeftIcon } from './icons/left.svg';
+import { ReactComponent as RightIcon } from './icons/right.svg';
+import { ReactComponent as TopIcon } from './icons/top.svg';
+import Titlebar from './titlebar/Titlebar';
 import ToolBar from './toolbar/ToolBar';
-import CloseButton from './toolbar/CloseButton';
-import './App.css';
-
-const isLauncherHorizontal = edge => edge === ScreenEdge.TOP;
 
 export default () => {
+  const [options] = useOptions();
+  const defaultHeight = options ? options.defaultHeight : 75;
+  const defaultWidth = options ? options.defaultWidth : 50;
+  const isDockable = options ? options.customData.isDockable : false;
+  const [leftPosition, setLeft] = useState(0);
+  const [topPosition, setTop] = useState(0);
+  const [isHorizontal, setHorizontal] = useState(true);
   const [edge, windowActions] = useDockWindow(
     ScreenEdge.TOP,
     OpenfinApiHelpers.getCurrentWindowSync(),
     true,
-    { dockedWidth: 50, dockedHeight: 50 }
+    {
+      dockedWidth: isHorizontal ? defaultWidth : 80,
+      dockedHeight: defaultHeight
+    },
+    {
+      undockPosition: { left: leftPosition, top: topPosition },
+      undockSize: { width: 724, height: 88 }
+    }
   );
 
-  const prevEdgeRef = useRef();
+  const iconStyle = isHorizontal ? 'icon horizontal' : 'icon vertical';
```

Comment:
Just a suggestion for less repetition: 
`const iconStyle = `icon ${isHorizontal ? 'horizontal' : 'vertical'}`